packages/react-icon-provider/src/IconProvider.tsx

```diff
@@ -0,0 +1,23 @@
+import { IIconSubset } from '@fluentui/style-utilities';
+import * as React from 'react';
+import { IconProviderProps } from './IconProvider.types';
+
+/**
+ * Context for providing the window.
+ */
+export const IconContext = React.createContext<IconProviderProps>({
+  icons: { icons: {} },
+});
+
+/**
+ * Hook to access the Icon object. This can be overridden contextually using the `IconProvider`.
+ */
+export const useIcon = (): IIconSubset | undefined => React.useContext(IconContext).icons;
+
+/**
+ * Component to provide the icon override object contextually. This is useful when rendering a different
+ * Icon from the default svg Icon provided
```

Comment:
Might be clearer to say something like this: "Component to contextually override one or more of the default SVG icons."